From 797a60117309cd1e8da369d52086bc61d6de141d Mon Sep 17 00:00:00 2001
From: Karthikeyan K <karthikk@habana.ai>
Date: Mon, 20 Dec 2021 06:40:22 +0200
Subject: [PATCH] Revert "[SW-63186] gaudi2: security: set axprot for trace"

As per alignment with LKD, MSTR_IF's OVRD register should not be used
to set to "USER level".

This is because sub-blocks which should be secure won’t be able to work,
as MSTR_IF's OVRD bit takes precedence over the engine's internal PROT
configuration.

This reverts commit cd8631ff994feace835f7bf2819dfaac1d22681e.

Change-Id: If983e5f5e5c6a63c5f2a6f49001aba4faa6425ab
---

diff --git a/arch/gaudi2/common/security.c b/arch/gaudi2/common/security.c
index 9c8a2c3..8d04a90 100644
--- a/arch/gaudi2/common/security.c
+++ b/arch/gaudi2/common/security.c
@@ -38,7 +38,6 @@
 #include <pmmu_hbw_mstr_if_core_hbw_regs.h>
 #include <arc_farm_kdma_mstr_if_core_hbw_regs.h>
 #include <arc_farm_kdma_mstr_if_core_lbw_regs.h>
-#include <psoc_global_conf_regs.h>
 
 /*
  * instances_num : Number of instances of RR to program.
@@ -963,12 +962,6 @@
 	regmap_writel(AXPROT_SEC_VAL, mmARC_FARM_KDMA_MSTR_IF_CORE_LBW_AXPROT_OVRD);
 }
 
-static void preboot_secure_trace(void)
-{
-	/* Set trace to user level */
-	regmap_writel(PROT_OVR_USER, mmPSOC_GLOBAL_CONF_TRACE_AXPROT);
-}
-
 void preboot_security_init(void)
 {
 	preboot_secure_pcie();
@@ -980,7 +973,6 @@
 		preboot_secure_rot();
 		preboot_secure_pmmu();
 		preboot_secure_arc_farm();
-		preboot_secure_trace();
 		preboot_secure_lbw();
 		preboot_secure_hbw();
 		preboot_puts("Security enabled.\n");
